weblate/addons/git.py

```diff
@@ -60,6 +60,34 @@ def get_filenames(self, component):
                 languages[code].extend(translation.filenames)
         return languages
 
+    def get_squash_commit_message(self, repository, log_format, remote, filenames=None):
+        commit_message = self.instance.configuration.get("commit_message")
+
+        if not commit_message:
+            command = [
+                "log",
+                "--format={}".format(log_format),
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            commit_message = repository.execute(command)
+
+        if self.instance.configuration.get("append_trailers"):
+            command = [
+                "log",
+                "--format={}".format("%(trailers:separator)"),
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            trailers = repository.execute(command)
```

Comment:
I guess this will can include same entries several times, not sure how big deal that is.

Reply:
Removing trailers from the "commit body" made it easy to remove duplicates as well, so I went ahead and did that since I could see no semantic reason to have duplicate trailers.